tests/macros/utils.js

```diff
@@ -4,83 +4,269 @@
+ *           using `jest.fn()` directly.
+ *
+ *           To unmock a template result, simply call `unmockTemplate()`
+ *           with the same `name`.
+ *
+ * @property {(name: string) => boolean} unmockTemplate
+ *           Stops mocking the result of a `template()` function call.
+ *
+ *           **Returns:**
+ *           - `true` if a macro result has previously been
+ *             mocked using `mockTemplate()`.
+ *           - `false` otherwise.
+ *
+ * @property {(...args: any[]) => Promise<string>} call
+ *           Use this function to make test calls on the named macro, if applicable.
+ *           Its arguments become the arguments to the macro. It returns a promise.
+ */
+
+/**
+ * When we were doing mocha testing, we used this.macro to hold this.
+ * But Jest doesn't use the this object, so we just store the object here.
+ * @type {Macro}
+ */
 let macro = null;
 
+/**
+ * Asserts that value is truthy.
+ *
+ * @param {any} x The value.
+ */
 function assert(x) {
-    expect(x).toBe(true);
+    expect(x).toBeTruthy();
 }
 
+/**
+ * Asserts deep equality of actual and expected.
+ *
+ * @param {any} x Actual value.
+ * @param {any} y Potential expected value.
+ */
 assert.equal = (x, y) => {
     expect(x).toEqual(y);
 };
 
 assert.eventually = {
+    /**
+     * Asserts deep equality of actual and expected.
+     *
+     * @param {any|Promise<any>} x A promise which resolves to the actual value.
+     * @param {any} y Potential expected value.
+     */
     async equal(x, y) {
         expect(await x).toEqual(y);
     }
 };
 
+/**
+ * Asserts that list includes the supplied element.
+ *
+ * @template T Type of values in list.
+ * @param {string|T[]} list Container string or array.
+ * @param {string|T} element Potential value contained in the list.
+ */
 assert.include = (list, element) => {
     expect(list).toContain(element);
 };
+
+/**
+ * Asserts that value is true.
+ *
+ * @param {boolean} value Actual value.
+ */
 assert.isTrue = value => {
     expect(value).toEqual(true);
 };
+
+/**
+ * Asserts that value is false.
+ *
+ * @param {boolean} value Actual value.
+ */
 assert.isFalse = value => {
     expect(value).toEqual(false);
 };
+
+/**
+ * Asserts value is strictly greater than (>) floor.
+ *
+ * @param {number} value Actual value.
+ * @param {number} floor Minimum Potential expected value.
+ */
 assert.isAbove = (value, floor) => {
     expect(value).toBeGreaterThan(floor);
 };
+
+/**
+ * Asserts that value is an array.
+ *
+ * @param {any[]} value Actual value.
+ */
 assert.isArray = value => {
     expect(value).toBeInstanceOf(Array);
 };
+
+/**
+ * Asserts that value is an object of type 'Object'
+ * (as revealed by Object.prototype.toString).
+ *
+ * @param {object} value Actual value.
+ * @remarks The assertion does match subclassed objects.
+ */
 assert.isObject = value => {
     expect(value).toBeInstanceOf(Object);
 };
+
+/**
+ * Asserts that value is a function.
+ *
+ * @param {Function} value Actual value.
```

Comment:
So this raises interesting questions....  The point of a runtime assertion that something is a function presumably means that we don't know statically whether it is a function. So declaring the argument to statically be a function seems like it will cause trouble...

It doesn't look like you've answered my questions about who or what these comments are for. If these are just really detailed doc comments for programmers reading the code, then this is probably fine. But if some developers will be actually using type-checking tools that read these comments, then I worry this will cause problems for tests that use assert.isFunction

Reply:
In that case, I’ll convert it all to `any`.